crates/stdarch-gen/neon.spec

```diff
@@ -424,6 +424,92 @@ arm = vcge.s
 // we are missing float16x4_t:uint16x4_t, float16x8_t:uint16x8_t
 generate float32x2_t:uint32x2_t, float32x4_t:uint32x4_t
 
+/// Compare signed greater than or equal to zero
+name = vcgez
+multi_fn = fixed, c:in_t
+multi_fn = fixed_2, d:in_t
+multi_fn = simd_shr, e:, a, transmute(c)
+multi_fn = simd_xor, transmute(e), transmute(d)
```

Comment:
Why not just use `simd_ge` here?

Reply:
This is to be consistent with Clang's implementation. The following is the test I did in https://godbolt.org/:

    #include <arm_neon.h>
    int test() {
      return (int) vcgez_s32;
    }        

And the Output:

    define dso_local i32 @test() local_unnamed_addr #0 {
        ret i32 ptrtoint (<2 x i32> (<2 x i32>)* @vcgez_s32 to i32)
    }
    
    define internal <2 x i32> @vcgez_s32(<2 x i32> %0) #1 {
        %2 = ashr <2 x i32> %0, <i32 31, i32 31>
        %3 = xor <2 x i32> %2, <i32 -1, i32 -1>
        ret <2 x i32> %3
    }
    
    attributes #0 = { norecurse nounwind readnone uwtable willreturn "frame-pointer"="non-leaf" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="generic" "target-features"="+neon" }
    attributes #1 = { alwaysinline norecurse nounwind readnone uwtable willreturn "frame-pointer"="non-leaf" "min-legal-vector-width"="64" "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="generic" "target-features"="+neon" }

Reply:
If you compile with `-O0` you will see that Clang actually emits an `icmp sge`. LLVM optimizations are then turning this into a shift + xor.

Reply:
The url of godbolt is from here: https://github.com/rust-lang/stdarch/issues/148

Reply:
That usually works, but in this particular case it gives a different result because the IR is not the one generated by Clang directly: it is the IR after LLVM has run optimization passes that expand the `icmp eq` into shift and xor.

You can use `simd_ge` in Rust and it will produce the same IR as Clang.

Reply:
Umm.. That's right. If we use an implementation consistent with -O0, can we ensure that LLVM achieves the same optimization? If so, we should indeed use simd_ge IMO

Reply:
We run the same LLVM passes as Clang (mostly) so rustc will also transform `simd_ge` into a shift + xor.

Reply:
Thanks for explanation!